packages/talisman-ui/src/components/MysticalBackgroundV3/useCelestialArtifact.ts

```diff
@@ -1,95 +1,175 @@
+    //
+    // Then once that initial state has rendered (thanks to useLayoutEffect) we will begin our first
+    // transition to the second position & size & color by calling `update` here.
+    //
+    // For this first transition, `initialized` will be `false` - so we will get a slightly different
+    // transition timing function (ease-in-out instead of ease-out).
+    //
+    // After that, `update` will call itself every `duration` interval, up until we unmount and set `active` to false.
+    update()
 
     return () => {
-      clearInterval(interval)
+      active = false
     }
-  }, [config, duration, parentSize, color])
+  }, [generate, duration])
 
-  return characteristics
+  return artifact
+}
+
+export type ParentSize = {
+  width: number
+  height: number
 }
+
+type GenerateArtifactProps = CelestialArtifactProps & {
+  /** The duration between each artifact position, scale and color change */
+  duration: number
+
+  /** Whether or not the artifact has been initialized */
+  initialized: boolean
+}
+
+const generateArtifact = ({
+  config,
+  parentSize,
+  artifactIndex,
+  color: forceColor,
+  duration,
+  initialized,
+}: GenerateArtifactProps): CSSProperties => {
+  const maxSize = Math.min(parentSize.width, parentSize.height) * 2
+  const width = Math.round(maxSize * randBetween(config.radiusMin, config.radiusMax))
+  const height = Math.round(width * (1 - config.ellipsisRatio * Math.random()))
+
+  const x = Math.round(Math.random() * parentSize.width)
+  const y = Math.round(Math.random() * parentSize.height - Math.max(width / 2, height / 2) / 2)
+
+  const color = forceColor ?? Color.hsv(Math.random() * 360, 100, 100).hex()
+  const opacity = randBetween(config.opacityMin, config.opacityMax)
+  const background = artifactBg(color, opacity)
+
+  // This defines the base unit used in the `scale3d()` transform func.
+  //
+  // It sets the base `width` and `height` of the element, before scaling with `scale3d` to the desired `width` and `height`.
+  //
+  // Why are we using `scale3d` and not just animating the `width` and `height` properties?
+  // Because `scale3d` can be GPU-accelerated, and changing it doesn't cause the browser to repaint the page on every frame.
+  //
+  // On chrome we can set this to `1px`, and then if we want a 500x500px circle we could do `scale3d(500, 500, 1)`.
+  //
+  // However, on firefox the radial-gradient background is computed *before* scaling the element.
+  // So if we use `1px`, we'll have a radial-gradient scaled to `500px` in screen size but with the resolution of just one pixel!
+  //
+  // This would *not* be a very nice gradient to look at :^)
+  //
+  // So to fix this, we'll scale this value up a few orders of magnitude, to `1000px`.
+  // And then for our 500x500px circle we need to use `scale3d(0.5, 0.5, 1)`.
+  // Because `500` (the size we want) divided by `1000` (our base size, or backgroundResolution) is `0.5`.
+  const backgroundResolution = 1000 // in pixels squared
```

Comment:
TLDR:
We want to scale big thing down to smaller size, not small thing (1x1px) up to bigger size.

It doesn't matter either way in chrome, but in firefox it breaks the whole effect when you go from small to big.